src/ts/components/AppBar.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {Component} from 'react';
+import {Link} from 'react-router';
+import {AppBar} from 'material-ui';
+import Menu from 'components/Menu';
+
+// Scroll logic based on https://gist.github.com/Warry/4254579
+
+const barStyle = {
+  position: 'fixed',
+  transform: 'translate3d(0,0,0)',
+  MozTransform: 'translate3d(0,0,0)',
+  WebkitTransform: 'translate3d(0,0,0)'
+};
+
+const titleStyle = {
+  fontFamily: 'Junicode',
+  fontSize: 43,
+  color: '#AD000D',
+  padding: '3px 0 0 14px',
+  marginBottom: -3
+};
+
+export default class AppBarOwn extends Component<any, any> {
+
+  constructor(props) {
+    super(props);
+    this.state = {visible: false};
+    this.handleScroll = this.handleScroll.bind(this)
+  }
+
+  shouldComponentUpdate() {
+    return false;
+  }
+
+  componentDidMount() {
+    this.handleScroll();
+  }
+
+  componentWillUnmount() {
+    this.handleScroll = () => null;
+  }
+
+  handleScroll() {
+    if (this.state.visible && window.scrollY === 0) {
+      this.setState({visible: false});
+      this.forceUpdate();
```

Comment:
The idea is from [here](https://www.toptal.com/react/react-redux-and-immutablejs), maybe it's not the best case, I was trying to figure out something between presentational component and component that keeps its state (but rerender only in specific situation like `window.scrollY === 0`)